app/Http/Controllers/Admin/RoleController.php

```diff
@@ -0,0 +1,337 @@
+            ];
+
+            $messages = [
+                'name.required' => __('Role') . __('Required'),
+                'name.string' => __('Role') . __('String'),
+                'name.unique' => __('Role') . __('Unique'),
+                'display_name.required' => __('Display name') . __('Required'),
+                'display_name.string' => __('Display name') . __('String'),
+                'display_name.unique' => __('Display name') . __('Unique'),
+            ];
+
+            $validator = Validator::make($data, $rules, $messages);
+
+            if ($validator->fails()) {
+                return response()->json([
+                    'error' => 'valid',
+                    'message' => $validator->errors(),
+                ]);
+            } else {
+                $this->roleRepository->create($data);
+
+                DB::commit();
+
+                return response()->json([
+                    'error' => false,
+                    'message' => 'Success !',
+                ]);
+            }
+        } catch (Exception $e) {
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        return view('admin.roles.show', [
+            'role_id' => $id,
+        ]);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        DB::beginTransaction();
+
+        try {
+            $role = $this->roleRepository->findOrFail($id);
+
+            DB::commit();
+
+            return response()->json([
+                'error' => false,
+                'message' => 'Success !',
+                'role' => $role,
+            ]);
+        } catch (Exception $e) {
+            DB::rollback();
+
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $data = $request->all();
+
+        DB::beginTransaction();
+
+        try {
+            $rules = [
+                'name'          =>  'required|string',
+                'display_name'  =>  'required|string',
+            ];
+
+            $messages = [
+                'name.required'         =>  __('Role') . __('Required'),
+                'name.string'           =>  __('Role') . __('String'),
+                'display_name.required' =>  __('Display name') . __('Required'),
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/RoleController.php</br>Expected 1 space between "'name.string'" and double arrow; 11 found